app/main.py

```diff
@@ -86,129 +89,47 @@
+def battle(KnightsConfig):
 
-        if "hp" in arthur["potion"]["effect"]:
-            arthur["hp"] += arthur["potion"]["effect"]["hp"]
+    # Preaparations
+    # list of lists [name, power, hp, protection]
+    Knights = []
 
-    # mordred
-    mordred = knightsConfig["mordred"]
+    for k, v in KnightsConfig.items():
+        # list [name, power, hp, protection)
+        Knight = []
 
-    # apply armour
-    mordred["protection"] = 0
-    for a in mordred["armour"]:
-        mordred["protection"] += a["protection"]
+        knight = KnightsConfig[k]
 
-    # apply weapon
-    mordred["power"] += mordred["weapon"]["power"]
+        Knight.append(knight["name"])
 
-    # apply potion if exist
-    if mordred["potion"] is not None:
-        if "power" in mordred["potion"]["effect"]:
-            mordred["power"] += mordred["potion"]["effect"]["power"]
+        Knight.append(knight["power"] + knight["weapon"]["power"])
 
-        if "protection" in mordred["potion"]["effect"]:
-            mordred["protection"] += mordred["potion"]["effect"]["protection"]
+        Knight.append(knight["hp"])
 
-        if "hp" in mordred["potion"]["effect"]:
-            mordred["hp"] += mordred["potion"]["effect"]["hp"]
+        Knight.append(Armour.app_armour(knight["armour"]))
 
-    # red_knight
-    red_knight = knightsConfig["red_knight"]
-
-    # apply armour
-    red_knight["protection"] = 0
-    for a in red_knight["armour"]:
-        red_knight["protection"] += a["protection"]
-
-    # apply weapon
-    red_knight["power"] += red_knight["weapon"]["power"]
-
-    # apply potion if exist
-    if red_knight["potion"] is not None:
-        if "power" in red_knight["potion"]["effect"]:
-            red_knight["power"] += red_knight["potion"]["effect"]["power"]
-
-        if "protection" in red_knight["potion"]["effect"]:
-            red_knight["protection"] += red_knight["potion"]["effect"]["protection"]
-
-        if "hp" in red_knight["potion"]["effect"]:
-            red_knight["hp"] += red_knight["potion"]["effect"]["hp"]
+        if knight["potion"] is not None:
+            Knight[1], Knight[2], Knight[3] = Potion.app_potion(
+                knight["potion"], Knight[1], Knight[2], Knight[3])
 
+        Knights.append(Knight)
+        print(Knights)
     # -------------------------------------------------------------------------------
-    # BATTLE:
-
-    # 1 Lancelot vs Mordred:
-    lancelot["hp"] -= mordred["power"] - lancelot["protection"]
-    mordred["hp"] -= lancelot["power"] - mordred["protection"]
+    # list result hp_after_fighting
+    result = []
+    for i in range(2):
+        result.append(Knights[i][2] - Knights[i + 2][1] + Knights[i][3])
+        print(result)
+        result.append(Knights[i + 2][2] - Knights[i][1] + Knights[i + 2][3])
 
-    # check if someone fell in battle
-    if lancelot["hp"] <= 0:
-        lancelot["hp"] = 0
+    for j in range(4):
```

Comment:
uninformative loop, why 4?

Reply:
4, becouse in this case 4 knights. I can change this value

Reply:
create more informative loop